submissions/Test123423235/CarRacingGame/Renderer.java

```diff
@@ -0,0 +1,115 @@
+package CarRacingGame;
+
+public class Renderer {
+    //Attributes
+    private IRenderObject[] Store = new IRenderObject[0];
+
+    //Help-Methods  //Methods from Algorithm1 Stack-Exercise to expand and reduce the Array
+
+    /**
+     * Intern Method to reduce the array by deleting the position with index unload_index
+     * @param existing_array, occurring Object-array
+     * @param unload_index, the index of the spot to delete
+     * @return reduced_array
+     */
+    private IRenderObject[] reduce (IRenderObject[] existing_array, int unload_index)
+    {
+        IRenderObject[] reduced_array = new IRenderObject[existing_array.length - 1];      //Create new array with decreased length
+
+        int p = 0;
+        for (int i=0; i < reduced_array.length; i++)                   //Copy all values to the new array
+        {                                                              //p is used to skip the copy of the unloaded spot
+            if ( i == unload_index)
+            {
+                p = i +1;
+            }
+            reduced_array[i] = existing_array[p];
+            p++;
+        }
+        return reduced_array;
+    }
+
+    /**
+     * Intern Method to expand the array and write down the attachment on the new extended place
+     * @param existing_array, occurring Object-array
+     * @param attachment, the Object to add
+     * @return expanded array
+     */
+    private IRenderObject[] expand (IRenderObject[] existing_array, IRenderObject attachment)
+    {
+        IRenderObject[] expanded_array = new IRenderObject[existing_array.length + 1];      //Create new array with increased length
+
+        for (int i=0; i < existing_array.length; i++)                   //Copy all values to the new array
+        {
+            expanded_array[i] = existing_array[i];
+        }
+        //Copy alternative
+        // if (expanded_array.length >= 0) System.arraycopy(existing_array, 0, expanded_array, 0, expanded_array.length);
+
+        expanded_array[existing_array.length] = attachment;             //Fill the extended spot with the attachment
+        return expanded_array;
+    }
+
+    //Methods
+    //Exception just for training reasons
+
+    public boolean addRenderObject(IRenderObject attachment){
+
+        try{
+            if (attachment == null){
+                throw new NullIRenderObjectException();
+            }
+        }catch (NullIRenderObjectException e){
+            //e.printStackTrace();
+            e.getMessage();
+        }
+
+        int stored = -1;
+        for ( int i = 0; i < Store.length; i ++){
```

Comment:
> If the `object` is already in the array (`equals`) do nothing and return `false`.

Possible solution: This check can be added to the for loop and move the actual assignment after the for loop. e.g. in the else case of the if